### PR TITLE
[Jenkins Source] - Change user and token to be optional

### PR DESCRIPTION
**Pull request by konfortes**
Jenkins instances can be configured to expose jobs and builds to unauthenticated users

## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## Related issues

> Fix [#870](https://github.com/faros-ai/airbyte-connectors/issues/870) 

## Migration notes

> No migrations

## Extra info

> Add any additional information

### CODE REVIEW

**Comment by konfortes** [comment]
recheck

File: sources/jenkins-source/src/jenkins.ts [hunk 1]
```diff
@@ -99,8 +99,10 @@ export class Jenkins {
       throw new VError('server_url: must be a valid url');
     }
 
-    jenkinsUrl.username = config.user;
-    jenkinsUrl.password = config.token;
+    if (config.user && config.token) {
```
Comment (tovbinm) [inline sources/jenkins-source/src/jenkins.ts:102]: I think we should error if one of them is set and other isnt

**Review by tovbinm — COMMENTED** [review]
you also need to remove the requirement from here - https://github.com/faros-ai/airbyte-connectors/blob/main/sources/jenkins-source/resources/spec.json#L7

File: sources/jenkins-source/src/jenkins.ts (hunk 1 above)
Comment (konfortes) [reply sources/jenkins-source/src/jenkins.ts:102]: sounds reasonable, will add it

File: sources/jenkins-source/src/jenkins.ts [hunk 2]
```diff
@@ -61,6 +61,34 @@ export class Jenkins {
...
   }
 
+  private static validateConfig(config: JenkinsConfig): void {
+    if (typeof config.server_url !== 'string') {
+      throw new VError('server_url: must be a string');
+    }
+
+    if (config.user || config.token) {
+      if (!config.user || !config.token) {
+        throw new VError('user and token must be both specified or both empty');
```
Comment (tovbinm) [inline sources/jenkins-source/src/jenkins.ts:71]: `user and token must be either both specified or both be empty`

**Review by tovbinm — APPROVED** [review]

File: sources/jenkins-source/src/jenkins.ts (hunk 2 above)
Comment (konfortes) [reply sources/jenkins-source/src/jenkins.ts]: Thanks, Fixed

**Review by tovbinm — APPROVED** [review]

**Comment by tovbinm** [comment]
Please resolve conflicts so we could merge

**Comment by konfortes** [comment]
@tovbinm conflicts resolved

**Review by tovbinm — APPROVED** [review]

**Pull request merged**